discovery/syncer.go

```diff
@@ -1105,3 +1116,21 @@ func (g *gossipSyncer) SyncState() syncerState {
 func (g *gossipSyncer) SyncType() syncerType {
 	return syncerType(atomic.LoadUint32(&g.syncType))
 }
+
+// ResetSyncedSignal returns a channel that will be closed in order to serve as
+// a signal for when the gossipSyncer has reached its chansSynced state.
+func (g *gossipSyncer) ResetSyncedSignal() chan struct{} {
+	g.Lock()
+	defer g.Unlock()
+
+	syncedSignal := make(chan struct{})
+
+	syncState := syncerState(atomic.LoadUint32(&g.state))
+	if syncState == chansSynced {
+		close(syncedSignal)
```

Comment:
also nil `g.syncedSignal`?

Reply:
Don't think there's a need to? If it's already set, then it'll be `close`d and `nil`d once reaching a `chansSynced` state.